python/lsst/cp/pipe/cpCtiSolve.py

```diff
@@ -611,3 +747,207 @@ def findTraps(self, inputMeasurements, calib, detector):
+
+
+        Parameters
+        ----------
+        signalVec : `np.ndarray`
+            Signal values for the dataset. Must be sorted
+            in ascending order.
+        dataVec : `np.ndarray`
+            Data values for the dataset. Must be sorted
+            in ascending order.
+        ctiRange : `list` [`float`]
+            Range of CTI within which to search for the
+            turnoff point.
+
+        Returns
+        -------
+        turnoff : `float`
+            The turnoff point in the same units as the
+            input signals
+        turnoffSamplingError : `float`
+            The sampling error of the turnoff point, equals
+            turnoff when not enough data points.
+
+        Notes
+        ------
+        If the data is sparse and does not cover the turnoff region,
+        it will likely just return the highest signal in the dataset.
+
+        However, it will issue a warning if the turnoff point is at
+        the edge of the search range.
+        """
+        # First, trim the data
+        idxs = (dataVec >= ctiRange[0]) * (dataVec <= ctiRange[1])
+        dataVec = dataVec[idxs]
+        signalVec = signalVec[idxs]
+
+        # Check for remaining data points
+        if dataVec.size == 0:
+            self.log.warning("No data points after cti range cut to compute turnoff "
+                             f"for amplifier {amp.getName()}. Setting turnoff point "
+                             "to 0 el.")
+            return 0.0, 0.0
+
+        if dataVec.size < 2:
+            self.log.warning("Insufficient data points after cti range cut to compute turnoff "
+                             f"for amplifier {amp.getName()}. Setting turnoff point "
+                             "to the maximum signal value.")
+            return signalVec[-1], signalVec[-1]
+
+        # Detrend the data
+        # We will use np.gradient since this method of
+        # detrending turns out to be more practical
+        # than using np.diff, which tends to be noisier.
+        # Besides, this tends to filter out the low
+        # gradient features of the data, particularly
+        # in the parallel turnoff.
+        detrendedDataVec = np.gradient(dataVec)
+
+        # Sigma clip the data to remove the
+        # turnoff points
+        cleanDataVecMaArray = sigma_clip(
+            detrendedDataVec,
+            sigma=self.config.turnoffFinderSigmaClip,
+            maxiters=self.config.turnoffFinderSigmaClipMaxIters,
+            cenfunc=np.nanmedian,
+            stdfunc=np.nanstd,
+            masked=True,
+        )
+
+        # Retrieve the result
+        good = ~np.ma.getmask(cleanDataVecMaArray)
+        cleanDataVec = np.ma.getdata(cleanDataVecMaArray)
+
+        if cleanDataVec.size == 0:
+            self.log.warning("No data points after sigma clipping to compute turnoff "
+                             f" for amplifier {amp.getName()}. Setting turnoff point "
+                             "to 0 el.")
```

Comment:
el -> electron?

Reply:
I was about to leave a comment asking these to be fully spelled out as well.